### PR TITLE
minipro: init at 0.5

### PR DESCRIPTION
**Pull request by bmwalters**
###### Motivation for this change

minipro is a useful command line utility for controlling TL866xx chip programmers.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/minipro/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  };
+
+  nativeBuildInputs = [ pkg-config which ];
+  buildInputs = [ libusb1 ];
+  makeFlags = [ "DESTDIR=$(out)" "PREFIX=" "CC=cc" ];
+
+  meta = with lib; {
+    homepage = "https://gitlab.com/DavidGriffith/minipro";
+    description = "An open source program for controlling the MiniPRO TL866xx series of chip programmers";
+    license = licenses.gpl3;
```
Comment (fabaff) [inline pkgs/tools/misc/minipro/default.nix:27]: The license looks like [`GPLv3+`](https://gitlab.com/DavidGriffith/minipro/-/blob/master/ihex.c#L6-9).

```suggestion
    license = licenses.gpl3Plus;
```

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -1316,6 +1316,12 @@
     github = "bmilanov";
     githubId = 30090366;
   };
+  bmwalters = {
```
Comment (fabaff) [inline maintainers/maintainer-list.nix:1319]: Can you please make this change in a desperate commit in this PR?

File: pkgs/tools/misc/minipro/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    maintainers = [ maintainers.bmwalters ];
+    platforms = platforms.all;
```
Comment (fabaff) [inline pkgs/tools/misc/minipro/default.nix:29]: Can be removed. This is the default.

**Review by fabaff — COMMENTED** [review]
The upstream repo contains [udev](https://gitlab.com/DavidGriffith/minipro/-/tree/master/udev) rules and they show up in the `Makefile`. BUt as far as I can tell are they not part of the result.

File: pkgs/tools/misc/minipro/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  src = fetchFromGitLab {
+    owner = "DavidGriffith";
+    repo = "minipro";
+    rev = version;
+    sha256 = "sha256-Hyj2LyY7W8opjigH+QLHHbDyelC0LMgGgdN+u3nNoJc=";
+  };
+
+  nativeBuildInputs = [ pkg-config which ];
+  buildInputs = [ libusb1 ];
+  makeFlags = [ "DESTDIR=$(out)" "PREFIX=" "CC=cc" ];
```
Comment (baloo) [inline pkgs/tools/misc/minipro/default.nix:22]: setting `PREFIX=` empty removes the install of udev rules, which are convenient to get the programmer accessible from a user.

Comment (baloo) [reply pkgs/tools/misc/minipro/default.nix:22]: by coincidence I made another derivation at the same time and I found:
https://github.com/NixOS/nixpkgs/pull/121836/files#diff-474178633c840f49ad63ac5037ec42ac6053ce906ad33b93f7a922aa21e6b327R21

to work pretty nicely.

**Comment by baloo** [comment]
@bmwalters do you have time to fix this? Otherwise I'd be happy to take over with the branch I pushed at #121836.

File: pkgs/tools/misc/minipro/default.nix (hunk 4 above)
Comment (bmwalters) [reply pkgs/tools/misc/minipro/default.nix]: Thanks for the feedback. I committed a slightly modified version to pick up the udev rules and bash completions.

**Comment by baloo** [comment]
LGTM

**Pull request merged**